activation/handler.go

```diff
@@ -281,56 +277,22 @@ func (h *Handler) handleAtx(
 		return nil, fmt.Errorf("%w: atx want %s, got %s", errWrongHash, expHash.ShortString(), id.ShortString())
 	}
 
-	// Check if processing is already in progress
-	h.inProgressMu.Lock()
-	if sub, ok := h.inProgress[id]; ok {
-		ch := make(chan error, 1)
-		h.inProgress[id] = append(sub, ch)
-		h.inProgressMu.Unlock()
-		h.logger.Debug("atx is already being processed. waiting for result",
-			log.ZContext(ctx),
-			zap.Stringer("atx_id", id),
-		)
-		select {
-		case err := <-ch:
-			h.logger.Debug("atx processed in other task",
-				log.ZContext(ctx),
-				zap.Stringer("atx_id", id),
-				zap.Error(err),
-			)
-			return nil, err
-		case <-ctx.Done():
-			return nil, ctx.Err()
+	key := string(id.Bytes())
+	proof, err, _ := h.inProgress.Do(key, func() (any, error) {
+		h.logger.Info("handling incoming atx", log.ZContext(ctx), zap.Stringer("atx_id", id), zap.Int("size", len(msg)))
```

Comment:
If I understand correctly, this is going to be called _a lot_ of times which isn't really clear whether it is needed or not. I would put it in `Debug` or remove it altogether, just because I'm not sure if it really gives value (certainly not to puny humans aka non-devs that may run their nodes in `Info` log-level

Reply:
I will change it to DEBUG as it's still valuable when debugging tests.

> (remember it would still cause allocations even if not logged).

It's not causing any allocations AFAIR. The `log.` and `zap.` functions return a `zapcore.Field` which lives on the stack.